sagemaker/modelmonitor.ipynb

```diff
@@ -0,0 +1,413 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Model Monitor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm=DkubeModelmonitor(name=MONITOR_NAME)\n",
+    "mt=ModelType.Regression\n",
+    "dc_t=DatasetClass.Train"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "text_file = open(\"transform-data.py\", \"r\")\n",
+    "#read whole file to a string\n",
+    "script = text_file.read()\n",
+    "#close file\n",
+    "text_file.close()\n",
+    "\n",
+    "with open('thresholds.json') as f:\n",
+    "    thresholds = json.load(f)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm.update_modelmonitor(model_type=str(mt),data_timezone=\"utc\",input_data_type=\"tabular\",thresholds=thresholds)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Training Details"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training_data = f'{DKUBE_BASE_DATASET}:'+ DKUBEUSERNAME\n",
+    "train_data_version = 'v1:'+api.get_dataset_versions(DKUBEUSERNAME,DKUBE_BASE_DATASET)[0]['version']['uuid']\n",
+    "prediction_data = MONITOR_NAME +'-predict:'+ DKUBEUSERNAME\n",
+    "labelled_data = MONITOR_NAME +'-groundtruth:'+ DKUBEUSERNAME\n",
+    "predict_data_format = str(DatasetFormat.Sagemakerlogs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm.add_datasources(data_class=str(DatasetClass.Train),name=training_data,data_format=str(DatasetFormat.Tabular),version=train_data_version,transformer_script = script)\n",
+    "mm.add_datasources(data_class=str(DatasetClass.Predict),name=prediction_data,data_format=predict_data_format,date_suffix=\"yyyy/mm/dd/hh\")\n",
+    "mm.add_datasources(data_class=str(DatasetClass.Labelled),name=labelled_data,data_format=str(DatasetFormat.Tabular),predict_col=\"charges\",groundtruth_col=\"GT_target\",timestamp_col=\"timestamp\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Add Drift monitoring details"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm.update_drift_monitoring_details(enabled=True,frequency=5,algorithm='auto')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create Model monitor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "api.modelmonitor_create(mm,wait_for_completion=True)"
```

Comment:
this should return the UUID of the monitor. You don't need to make another call below

Reply:
done in PR https://github.com/oneconvergence/dkube/pull/103